consumer.go

```diff
@@ -206,6 +208,10 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
 			}
 
 			shardIterator = resp.NextShardIterator
+			if c.immediateRescan && len(resp.Records) > 0 && *resp.MillisBehindLatest > 0 {
+				// There appears to be additional records, so skip waiting for next tick
+				continue
+			}
 		}
```

Comment:
is there a situation where we wouldn't want immediate rescan? curious about pulling this down here:

```go
for ... {
}

len(resp.Records) > 0 {
  continue
}
```

note: I didn't add the ticker between scans. it felt ok to me, but now reading this I don't see the case where I wouldn't want the new records immediately

Reply:
I don't either :) I made it optional so as not to change existing behavior, but if you think (as I do) this is desirable default behavior I can remove the option.

I included the check of resp.MillisBehindLatest to deal with the fact that every scan that yielded records generated a flurry punctuated with a call that returned 0 records.  Checking this eliminated superfluous calls while still improving the responsiveness.